core/src/main/scala/org/apache/spark/deploy/rest/StandaloneRestServer.scala

```diff
@@ -124,7 +124,10 @@ private[rest] class StandaloneSubmitRequestServlet(
    * fields used by python applications since python is not supported in standalone
    * cluster mode yet.
    */
-  private def buildDriverDescription(request: CreateSubmissionRequest): DriverDescription = {
+  private[rest] def buildDriverDescription(
+      request: CreateSubmissionRequest,
+      masterUrl: String,
+      masterRestPort: Int): DriverDescription = {
```

Comment:
This private method is revised for better testing by using dependency injection pattern.